vowpalwabbit/core/src/reductions/eigen_memory_tree.cc

```diff
@@ -0,0 +1,787 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw/core/reductions/eigen_memory_tree.h"
+
+#include "vw/common/future_compat.h"
+#include "vw/config/options.h"
+#include "vw/core/array_parameters.h"
+#include "vw/core/example.h"
+#include "vw/core/learner.h"
+#include "vw/core/numeric_casts.h"
+#include "vw/core/rand48.h"
+#include "vw/core/rand_state.h"
+#include "vw/core/setup_base.h"
+#include "vw/core/v_array.h"
+#include "vw/core/vw.h"
+#include "vw/io/logger.h"
+
+#include <algorithm>
+#include <cfloat>
+#include <climits>
+#include <cmath>
+#include <cstdio>
+#include <ctime>
+#include <list>
+#include <memory>
+#include <sstream>
+#include <type_traits>
+
+using namespace VW::LEARNER;
+using namespace VW::config;
+
+namespace VW
+{
+namespace reductions
+{
+namespace eigen_memory_tree
+{
+////////////////////////////definitions/////////////////////
+////////////////////////////////////////////////////////////
+emt_example::emt_example() { label = 0; }
```

Comment:
IMO, just set label to 0 in the class definition and then don't define a constructor and let the compiler do it

Reply:
@jackgerrits I don't think I can do this because there is another constructor defined on this class.

Reply:
In that case you can just mark it in the header like so:
```c++
struct emt_example
{
  emt_feats base;  // base example only includes the base features without interaction flags
  emt_feats full;  // full example includes the interactions that were passed in as flags
  uint32_t label = 0;

  emt_example() = default // <-----------
  emt_example(VW::workspace&, VW::example*);
};

```

Reply:
ooh fancy, I like it